src/main/java/dudu/controller/FileOperation.java

```diff
@@ -0,0 +1,152 @@
+package dudu.controller;
+
+import dudu.model.Task;
+import dudu.model.TaskList;
+import dudu.model.Todo;
+import dudu.model.Deadline;
+import dudu.model.Event;
+
+import dudu.utils.DateTimeParser;
+import dudu.utils.DuduException;
+import dudu.utils.Ui;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.Scanner;
+
+public class FileOperation {
+
+    /**
+     * Returns the local save file.
+     * If no local save file is found, create a new one.
+     *
+     * @return Save file.
+     * @throws IOException If error occurs while retrieving the file.
+     */
+    public static File loadFile() throws IOException {
+        File file = new File("./data/duduTasks.txt");
+
+        file.getParentFile().mkdirs();
+
+        if (file.createNewFile()) {
+            Ui.printContent("No tasks file found. Creating a new tasks file at: " + file.getAbsolutePath());
+        } else {
+            Ui.printContent("Cached tasks loaded.");
+        }
+
+        return file;
+    }
+
+    /**
+     * Parses a line from the save file and adds it to the current list of tasks.
+     *
+     * @param line Line currently being read from save file.
+     * @return Appropriate Task object.
+     * @throws DuduException If invalid data is received from the save file.
+     */
+    public static Task parseFile(String line) throws DuduException {
+        String[] parts = line.split("\\|");
+
+        for (int i = 0; i < parts.length; i++) {
+            parts[i] = parts[i].trim();
+        }
+
+        String task = parts[0];
+        String done = parts[1];
```

Comment:
Consider using a boolean `isDone` instead of repeating the same check in every case.
```suggestion
        boolean isDone = parts[1].equals("Done");
```